### ISSUE TITLE
Some questions about resnet

### ISSUE THREAD

**Issue by shihaobai** [WenbinLee/DN4#14]
Very nice job!!! I tried to use resnet as a backbone just like what you did in  my own experiment.  However, it didn't improve performance as we expected and even performed worse than 4-conv. I wonder if there are some tricks when you train your model with a resnet. Thank you very much.

**Comment by WenbinLee** [comment]
Thanks. Actually, I didn't use specific tricks and just replaced 4-Conv backbone with Resnet backbone in our experiments. I am sure the true reason in your experiments. I guess the issue may be caused by the __pycache__ file. Maybe you can try to delete all the __pycache__ files and add "import sys   sys.dont_write_bytecode = True" in each script to avoid generating __pycache__ files.

**Comment by shihaobai** [comment]
Thanks for your relpy! I have tried to remove the pycache files but it didn't work. . The accuracy in training step reached to 1.0 very quickly so that the performance is worse in test step. Therefore, I wonder if you have encountered a similar situation and how you look at this phenomenon.Thank you very much.

**Comment by WenbinLee** [comment]
It's my pleasure. Yeah, the situation will happen when the dataset is very small but the backbone is too deep. So what's dataset you used in your experiment and how deep the ResNet you employed?
To overcome this situation, you may have the following solutions: (1) try data augmentation; (2) use the pre-trained model (can only using the auxiliary set); (3) use a small ResNet (e.g., ResNet 9) ;(4) use a large image size (e.g., 84* 84 or 224*224, but should be fair) ; (5) use a smaller learning rate. Hope this helps. Good luck!

**Comment by shihaobai** [comment]
Thanks for your help!!! Using pre-trained model did help slow the overfitting. I just trained my resnet with training set of miniImageNet. So what's the auxiliary set? Could you share it download link? Thank you for your patient reply very much!!

**Comment by WenbinLee** [comment]
You are welcome. Actually, the auxiliary set is just the training set of the miniImageNet. It does easily make people confused about the settings of few-shot learning. We usually employ an auxiliary set (64 classes) as the training set and employ another set as the test set (20 classes). As for the few-shot task, e.g., a 5-way 1-shot task, we can randomly construct such one task from the training set or test set. You can read our paper for more details.

**Comment by shihaobai** [comment]
Ok! Thank you again!
